test/distributed/_tensor/test_device_mesh.py

```diff
@@ -239,7 +240,8 @@ def world_size(self):
     def test_all_reduce_1d(self):
         mesh = DeviceMesh(self.device_type, torch.arange(self.world_size))
         local_tensor = torch.ones(3, 3, device=self.device_type) * self.rank
-        mesh.all_reduce(local_tensor, mesh_dim=0)
+        # multiply with 1 to trigger wait
+        local_tensor = mesh.all_reduce(local_tensor, mesh_dim=0) * 1
```

Comment:
i'm surprised you need to multiply with 1 to trigger wait.

i expected the assert statement would trigger some op on the tensor that forced the wait.

Reply:
Not really, go figure :/